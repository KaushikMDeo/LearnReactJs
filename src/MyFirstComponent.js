import React from 'react';

const MyFirstComponent = React.createClass({
  render() {
    return (
      <h1>Hello, world!</h1>
    );
  }
});

export default MyFirstComponent;
