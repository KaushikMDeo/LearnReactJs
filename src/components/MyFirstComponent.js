import React, { PropTypes } from 'react';
import styles from './MyFirstComponent.css';

const MyFirstComponent = React.createClass({
  propTypes: {
    onClose: PropTypes.func.isRequired,
    showSmallHeader: PropTypes.bool
  },
  render() {
    const { onClose, showSmallHeader } = this.props;
    return (
      <div>
        <div className="modal-header">
          <button type="button" className="close" aria-label="Close" onClick={onClose}><span aria-hidden="true">&times;</span></button>
          <h4 className="modal-title">My First Component 123</h4>
        </div>
        <div className="modal-body">
          <h1 className={styles.blue}>Hello, world!</h1>
          {showSmallHeader ? <h2>Hello, minions!</h2> : null}
        </div>
        <div className="modal-footer">
          <button type="button" className="btn btn-primary" onClick={onClose}>Close</button>
        </div>
      </div>
    );
  }
});

export default MyFirstComponent;
