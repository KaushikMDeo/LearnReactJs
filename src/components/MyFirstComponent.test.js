import React from 'react';
import MyFirstComponent from './MyFirstComponent';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <MyFirstComponent onClose={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders small header', () => {
  const tree = renderer.create(
    <MyFirstComponent onClose={() => {}} showSmallHeader={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders hide small header', () => {
  const tree = renderer.create(
    <MyFirstComponent onClose={() => {}} showSmallHeader={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
