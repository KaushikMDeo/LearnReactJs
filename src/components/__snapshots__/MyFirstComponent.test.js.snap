// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      My First Component 123
    </h4>
  </div>
  <div
    className="modal-body"
  >
    <h1
      className="blue"
    >
      Hello, world!
    </h1>
  </div>
  <div
    className="modal-footer"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`renders hide small header 1`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      My First Component 123
    </h4>
  </div>
  <div
    className="modal-body"
  >
    <h1
      className="blue"
    >
      Hello, world!
    </h1>
  </div>
  <div
    className="modal-footer"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`renders small header 1`] = `
<div>
  <div
    className="modal-header"
  >
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    <h4
      className="modal-title"
    >
      My First Component 123
    </h4>
  </div>
  <div
    className="modal-body"
  >
    <h1
      className="blue"
    >
      Hello, world!
    </h1>
    <h2>
      Hello, minions!
    </h2>
  </div>
  <div
    className="modal-footer"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
